homeassistant/components/tradfri/cover.py

```diff
@@ -5,6 +5,12 @@
 from .base_class import TradfriBaseDevice
 from .const import CONF_GATEWAY_ID, KEY_API, KEY_GATEWAY
 
+ATTRIBUTE_ALIAS = {
+    "battery_level": None,
```

Comment:
Battery level should be its own sensor, with device class "battery"

Reply:
There are other integrations that have the battery_level as attributes ... and personally I prefer them with the device itself where possible.

Reply:
Yes, there are many old examples that don't follow our current design guidelines. Future changes should follow them though.

Reply:
I know that this is a little off topic ... but
Personally I think that the old way is better then.

Why ... Because I can make a Lovelace card that extends entities in an entities card to automatically add battery levels to everything that has a battery level so it appears like so:
![image](https://user-images.githubusercontent.com/1435262/78464349-9f695180-772b-11ea-8d20-51bf64ff3c34.png)
... good luck matching batteries to entities if they are separate entities themselves.

... This is in my production one (not development) blinds also have the Battery with this change.

Reply:
.. Also doesn't it make more sense to have the battery info with the device it is in?..
... imagine this applying this level of extraction to all of the attributes in a vac for example (would be a lot of individual sensors for the same thing):
```yaml
battery_level: 100
battery_icon: 'mdi:battery-charging-100'
fan_speed: 104
do_not_disturb: 'off'
do_not_disturb_start: '22:00:00'
do_not_disturb_end: '08:00:00'
cleaning_time: 14
cleaned_area: 13
cleaning_count: 24
total_cleaned_area: 205
total_cleaning_time: 207
main_brush_left: 296
side_brush_left: 196
filter_left: 146
sensor_dirty_left: 26
status: Charging
clean_start: '2020-04-05T11:46:11'
clean_stop: '2020-04-05T12:00:18'
friendly_name: Living room vacuum
supported_features: 14204
coverage_area_max: 16
```

Reply:
Also where can I find the current design guidelines that mention this? ... I am quite sure that this is not explicitly mentioned in the developer documentation... the only place that anything like this is mentioned is written as an example: "for example, a battery-powered temperature and humidity sensor might expose entities for temperature, humidity and battery level." (https://developers.home-assistant.io/docs/device_registry_index/); however the way that this is written seems to suggest that it is a demonstration of how things _can_ be done.

Thanks

Reply:
Sorry if I came across a little strong ... but I have just been looking through the 2 battery entities that I have that are separate... and just noticed that the [Home Assistant Mobile App Support](https://www.home-assistant.io/integrations/mobile_app/) integration has the battery as an attribute as well as a separate entity; I am not opposed to doing this as it would allow people to query the battery using the method of their choice.. How can I add the battery as an entity to the cover device (I have not done anything with devices before and am fairly new to programming HA integrations)?

Thanks, Kris

Reply:
Adding another entity should be a separate PR.
As for a battery sensor to the blinds, make sure you test that over time.
Last time I tried adding that it resulted in a lot of instability in the integration due to how the listeners works.

Reply:
Do connect a battery sensor entity to a tradfri (cover) device, set the correct device id in the `device_info` entity property. This will tell Home Assistant that the entity belongs to the device in question.

Reply:
@MartinHjelmare ... thanks ... will look at this after I finish work 👍